### PR TITLE
GL ES detect

### PR DESCRIPTION
**Pull request by daschuer**
This is an attempt to fix the blank GL Status reported here: 
https://bugs.launchpad.net/mixxx/+bug/1825461

### CODE REVIEW

**Comment by uklotzde** [comment]
Should we remove all the remaining obsolete QT_VERSION_CHECK guards up to and including Qt 5.2 in a separate PR? I noticed that there are many of them that unnecessarily clutter the code.

**Comment by daschuer** [comment]
> Should we remove all the remaining obsolete QT_VERSION_CHECK guards up to and including Qt 5.2 in a separate PR? I noticed that there are many of them that unnecessarily clutter the code.

Yes, probably along with the string based connect() calls. The later hurts me much more.

**Comment by daschuer** [comment]
Ok, it is working. My virtualized Windows 10 is now reporting OpenGL ES 2.0. All GL and GLS waveforms are removed from the combobox.

I guess it is because I have messed arround with Qt fall back renderers. Or something.

**Comment by daschuer** [comment]
Interestingly spinnies and Qt waveforms are  working in this state. 
So it is not correct to hide the Qt waveforms  now.

**Comment by uklotzde** [comment]
LGTM, although I'm not able to actually test it. Ready to merge?

**Comment by daschuer** [comment]
Not ready, Qt5 requires at least OpenGL 2.1 or OpenGL ES 2.0. I will update the versions check accordingly.

**Comment by daschuer** [comment]
Now the GL detection should match QTs ideas of using GL in QT5. 
It also fixes a crasher related to spinnies: https://bugs.launchpad.net/mixxx/+bug/1798858

**Comment by daschuer** [comment]
Maybe it fixes also the cash reported in https://bugs.launchpad.net/mixxx/+bug/1341036

**Comment by daschuer** [comment]
This PR is also a base for GL ES waveforms: 
https://bugs.launchpad.net/mixxx/+bug/1316402

File: src/waveform/waveformwidgetfactory.cpp [hunk 1]
```diff
@@ -667,98 +721,129 @@ void WaveformWidgetFactory::evaluateWidgets() {
...
             widgetName = GLVSyncTestWidget::getWaveformWidgetName();
             useOpenGl = GLVSyncTestWidget::useOpenGl();
+            useOpenGles =  GLVSyncTestWidget::useOpenGles();
             useOpenGLShaders = GLVSyncTestWidget::useOpenGLShaders();
             developerOnly = GLVSyncTestWidget::developerOnly();
             break;
         case WaveformWidgetType::GLRGBWaveform:
             widgetName = GLRGBWaveformWidget::getWaveformWidgetName();
             useOpenGl = GLRGBWaveformWidget::useOpenGl();
+            useOpenGles =  GLVSyncTestWidget::useOpenGles();
```
Comment (uklotzde) [inline src/waveform/waveformwidgetfactory.cpp]: Copy&Paste error?

**Comment by daschuer** [comment]
This one has the regression that there are no waveforms at all if you start Mixxx with: 
mixxx --developer 
on windows 10. 
Without the developer flag everything is fine.

**Comment by daschuer** [comment]
OK, the initialization of the VSyncThread was timing sensitive. This is now fixed. 
Hopefully that fixes https://bugs.launchpad.net/mixxx/+bug/1824620 as well.

**Comment by uklotzde** [comment]
Is this ready? I'm slightly confused by the two parallel PRs. Shouldn't the other be marked as WiP until this has been merged both into 2.2 and master?

**Comment by uklotzde** [comment]
Travis fails: QOpenGLFunctions::glGetString() was introduced in Qt 5.3!

**Review by uklotzde — CHANGES_REQUESTED** [review]
Requires Qt 5.3

**Review by uklotzde — APPROVED** [review]

**Comment by uklotzde** [comment]
LGTM. AppVeyor CI build is expected to succeed.

**Pull request merged**